### ISSUE TITLE
Finetune pretrained models on different dataset

### ISSUE THREAD

**Issue by matteopilotto** [mlmed/torchxrayvision#63]
Hi all and thank you for developing this library.
In the readme it's said that the pretrained models can be used for feature reuse for few-shot learning.
Does that mean that it's possible to take a pretrained model and finetune it for a couple of epochs on a different dataset?
I spent some time to explore the repo, but I couldn't find anything related to that.
I would truly grateful if you could help me on that.
Thanks

**Comment by ieee8023** [comment]
Sorry for the delay! I have an example that we are using in an upcoming publication. I'll make a working example and add it to the scripts folder today or tomorrow.

**Comment by ieee8023** [comment]
Here is an example of finetuning the classifier layer of a densenet on a dataset for lung opacity: https://github.com/mlmed/torchxrayvision/blob/master/scripts/transfer_learning.ipynb

Let us know if you have any questions or find any bugs!

There is also another project that does this but uses sklearn on extracted features here: https://github.com/mlmed/covid-severity but it is not a clean as the new example above.

**Comment by matteopilotto** [comment]
Hey @ieee8023,
Thanks for answering my question.
Your notebook works just fine with the default datasets. The possibility to use these pretrained models is awesome!
Since I'm most familiar with fastai, I've been trying to create a dataloader using that library, but unfortunately I cannot get meaningful results because I'm able to properly applied the same normalization you applied to your dataset.
That's why I would like to know if with this library is possible to directly create a dataset using custom data.
Thanks again for the help.

**Comment by ieee8023** [comment]
> unfortunately I cannot get meaningful results because I'm able to properly applied the same normalization you applied to your dataset.

You can process images like this: https://github.com/mlmed/torchxrayvision/blob/master/scripts/process_image.py#L29

Another thing that may work for you is using model.features2(img) on the DenseNet (https://github.com/mlmed/torchxrayvision/blob/master/torchxrayvision/models.py#L219). 

This will return a feature vector for the image that you can plug into another model. If you normalize the input image pixel between range [-1024,1024] and they are square it will work.

> That's why I would like to know if with this library is possible to directly create a dataset using custom data.

Ya, you can implement a dataset like the ones in the library but it doesn't matter to match their interface unless you are going to use the dataset tools. You just need to scale the images to feed them into the models. You can do it like we do in the provided datasets: https://github.com/mlmed/torchxrayvision/blob/master/torchxrayvision/datasets.py#L339

**Comment by ieee8023** [comment]
Closing this for now. Please reopen if the issue was not resolved.
